src/dbnode/storage/database.go

```diff
@@ -317,14 +309,41 @@ func (d *db) UpdateOwnedNamespaces(newNamespaces namespace.Map) error {
 			"restart the process if you want changes to take effect")
 	}
 
-	// enqueue bootstraps if new namespaces
 	if len(adds) > 0 {
-		d.queueBootstrapWithLock()
+		// NB: need to disable fileOps and wait for all the background processes to complete
+		// so that we could update namespaces safely. Otherwise, there is a high chance in getting
+		// invariant violation panic because cold/warm flush will receive new namespaces
+		// in the middle of their operations.
+		d.disableFileOpsAndWait()
+		d.Lock()
+		defer d.Unlock()
+		// add any namespaces marked for addition
+		if err := d.addNamespacesWithLock(adds); err != nil {
+			enrichedErr := fmt.Errorf("unable to add namespaces: %w", err)
+			d.log.Error(enrichedErr.Error())
+			d.enableFileOps()
+			return err
```

Comment:
I realize this was here but it must not have been converted properly, the zap way is to:
```suggestion
			d.log.Error("unable to add namespaces", zap.Error(err))
			d.enableFileOps()
			return err
```